riot_pal/dut_pyshell.py

```diff
@@ -0,0 +1,258 @@
+        map_records = self.cmd_list
+        return [s[offs:] for s in map_records if s.startswith(mline)]
+
+    def do_show_pinout(self, arg):
+        """Prints the pinout for the connected board
+
+        Usage:
+            show_pinout
+        """
+        print("""
+____________$$$$$$$$$$$$
+__________$$___________$$$
+_________$$_______________$$____________$$$$
+_______$$___________________$$______$$$$___$
+______$$___________________$$$$$___$$$____$
+_____$$__________________$$$___$$$$______$
+____$$________________$$$_____$__$_______$
+___$$_______________$$$____$$___$$$$____$
+___$$______________$$____$_____$$__$$___$
+___$$____________$$___$$_______$_$$$_$_$
+___$$___________$$_$$$_$$$$$____$$$$__$$
+___$$__________$_$$$$_$$__$$$$__$$$____$
+____$$________$$§§§§__$__$$$$$_____$$___$
+_____$$______$$§$§§$_____$$$_______$$$___$
+______$$___$$$§§$§§$$_____________________$
+_______$$_$$$§§$§§$§§$______________$$$$__ $
+_________$$$§§$$§$$§§$_____________$__$$__ $
+_______§$§$§§$$§§$§§§$____________$$$$$___ $
+______§$§$$§$§§§$$§§$_____________________ $
+_____$$§$$$$§§§$$$§$$$$___________________$
+____$$§$$$$§§§$$§$§§$$$$$$______________$$
+____$§§$§$§§§§$§§$§§$$$$___$$$$$$$$$$$$$
+___$$§$§§§§§§$§§§$§§$$$______________$
+___$§$$§§§§§$§§§§$§§$$$$____________$
+____$$§§§§§$§§§§§$§§$$$$____________$
+_____$§§§$$$§§§§§$§§$$$$$$$________$
+____$$§§$§§$§§§§$§§$$$$$$_$________$
+____$§§§§§§$§§§§$§§$$$_____$$_____$
+___$§§§§§§$§§§§$$§$$__________$$_$$
+$$$$§§§§§$$§§§§$§§$_____________$$
+$$§§§§§§$$$$§§$§§$$_______________$
+$§§§§$$§§§§§§$§§$__________________$$
+$§$$§§§§$§§$$§$$____________________$
+$$§$§§§$$§§$§$$$____________________$$
+$$$$§§§§§§§$$$$_____________________$$$
+§$$§§§$§§§§$$$$_____________________$$$
+__$$§§§§$§$§$_______________________$$$
+
+""")
+
+    def do_exit(self, arg):
+        """I mean it should be obvious
+
+        Usage:
+            exit
+        """
+        arg = arg
+        return True
+
+    def _print_func_result_success(self, results):
+        if not isinstance(results, list):
+            results = [results]
+        result = RESULT_SUCCESS
+        printed_something = False
+        for res in results:
+            if self.data_only:
+                if 'result' in res:
+                    if res['result'] == RESULT_SUCCESS:
+                        if 'data' in res:
+                            for data in res['data']:
+                                print(dumps(data))
+                                printed_something = True
+                    else:
+                        result = res['result']
+                else:
+                    print(dumps(res))
+                    printed_something = True
+            else:
+                print(dumps(res))
+                printed_something = True
+        if not printed_something:
+            print(result)
+
+    def _print_func_result(self, func, arg):
+        value = (arg or '').split(' ')
+        func_args = [v for v in value if v]
```

Comment:
can be omitted bc `value == func_args`, so simply use `value` (or better name it `values` or `func_args` directly.

Reply:
Does this filter out all empty values (if you have two spaces)? It was copy pasta... I will change the names and test.

Reply:
ah you're right, currently it does not but it works if you use
```Python
value = arg.split()
```

bc `split` without specifying the separator split by any number of spaces, and does not create empty items.